### ISSUE TITLE
Lookahead in Group

### ISSUE THREAD

**Issue by alvarofpp** [alvarofpp/mre#6]
**Lookahead assertion**
[`(?=...)`](https://docs.python.org/3/library/re.html#index-20)
```py
from mre import Regex, Group

# Example of how it can work 1
print(Regex("Isaac", Group(" Asimov", lookahead=True)))

# Output: Isaac(?= Asimov)
# Regex without lookahead: Isaac( Asimov)
```
![lookahead-positive](https://user-images.githubusercontent.com/10817238/65729929-7427ea80-e096-11e9-9c33-bd640ff7447e.png)

**Negative lookahead assertion**
[`(?!...)`](https://docs.python.org/3/library/re.html#index-21)
```py
from mre import Regex, Group

# Example of how it can work 1
print(Regex("Isaac", Group(" Asimov", lookahead=False)))

# Output: Isaac(?! Asimov)
# Regex without lookahead: Isaac( Asimov)
```
![lookahead-negative](https://user-images.githubusercontent.com/10817238/65729952-8a35ab00-e096-11e9-9382-74b4307007d8.png)

**Note**: the default value of the `lookahead` parameter is `None`.

**Comment by sgaist** [comment]
Hi,

I'd like to take on this one :-)

**Comment by sgaist** [comment]
I would say there's something wrong here in mixing Group with lookahead (and lookbehind for that mather). They are different things. Looking around in a regex is a non-capturing operation and has a different semantic than grouping although in uses the same symbol to start. I would recommend having a separated class for that.

**Comment by alvarofpp** [comment]
@sgaist 

I agree with you. I really didn’t try to do that. The classes could be created:

 - `LookaheadGroup` (or just `Lookahead`);
 - `LookbehindGroup` (or just `Lookbehind`).

 Both could inherit from `Group`, passing `non_capturing=False` in the parent class's constructor.
